src/rpc/connectivity/cluster.cc

```diff
@@ -953,7 +971,7 @@ void connectivity_cluster_t::run_t::handle(
                 }
             }
             fail_handshake(conn, peername, reason, handshake_error_supported);
-            return;
+            return join_result_t::TEMPORARY_ERROR;
```

Comment:
I think an incompatible protocol version should be a PERMANENT_ERROR. If the administrator restarts the other server later with a different version, that server is going to connect to us. So there is no need to keep trying to connect to it I think.